src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -26,29 +35,186 @@ const ActorInit Item_Etcetera_InitVars = {
+    ItemEtcetera* this = THIS;
+    s32 pad;
+    s32 type;
+    s32 objBankIndex;
+
+    type = this->actor.params & 0xFF;
+    osSyncPrintf("no = %d\n", type);
+    objBankIndex = Object_GetIndex(&globalCtx->objectCtx, sObjectIds[type]);
+    osSyncPrintf("bank_ID = %d\n", objBankIndex);
+    if (objBankIndex < 0) {
+        __assert("0", "../z_item_etcetera.c", 241);
+    } else {
+        this->objBankIndex = objBankIndex;
+    }
+    this->drawId = sDrawItemTableIdx[type];
+    this->getItemId = sGetEtceteraItemId[type];
+    this->actionFunc_14C = func_80B85824;
+    this->currentDrawRoutine = ItemEtcetera_DrawFireArrow;
+    Actor_SetScale(&this->actor, 0.25f);
+    ItemEtcetera_SetupAction(this, func_80B857D0);
+    switch (type) {
+        case ITEM_ETCETERA_LETTER:
+            Actor_SetScale(&this->actor, 0.5f);
+            this->actionFunc_14C = func_80B858B4;
+            if (gSaveContext.eventChkInf[3] & 2) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case ITEM_ETCETERA_ARROW_FIRE:
+            this->actionFunc_14C = ItemEtcetera_UpdateFireArrow;
+            Actor_SetScale(&this->actor, 0.5f);
+            this->actor.draw = NULL;
+            this->actor.shape.unk_08 = 50.0f;
+            break;
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+        case 12:
+        case 13: // unused values?
+            Actor_SetScale(&this->actor, 0.5f);
+            this->actionFunc_14C = func_80B85B28;
+            this->currentDrawRoutine = func_80B85C64;
+            this->actor.posRot.pos.y += 15.0f;
+            break;
+    }
+}
+
+void ItemEtcetera_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80B857D0(ItemEtcetera* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        this->actor.objBankIndex = this->objBankIndex;
+        this->actor.draw = this->currentDrawRoutine;
+        this->actionFunc_15C = this->actionFunc_14C;
+    }
+}
+
+void func_80B85824(ItemEtcetera* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx) != 0) {
+        if ((this->actor.params & 0xFF) == 1) {
+            gSaveContext.eventChkInf[3] |= 2;
+            Flags_SetSwitch(globalCtx, 0xB);
+        }
+        Actor_Kill(&this->actor);
+    } else {
+        func_8002F434(&this->actor, globalCtx, this->getItemId, 30.0f, 50.0f);
+    }
+}
+
+void func_80B858B4(ItemEtcetera* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx) != 0) {
+        if ((this->actor.params & 0xFF) == 1) {
+            gSaveContext.eventChkInf[3] |= 2;
+            Flags_SetSwitch(globalCtx, 0xB);
+        }
+        Actor_Kill(&this->actor);
+    } else {
+        if (0) {} // Necessary to match
+        func_8002F434(&this->actor, globalCtx, this->getItemId, 30.0f, 50.0f);
+        if ((globalCtx->gameplayFrames & 0xD) == 0) {
```

Comment:
think this is gonna be % 14 but could be wrong

Reply:
It messes everything up if i try to use mod here